src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/PoolingDataSource.cs

```diff
@@ -45,19 +90,246 @@ internal sealed class PoolingDataSource : SqlDataSource
         {
             _connectionPoolGroupOptions = options;
             _connectionRateLimiter = connectionRateLimiter;
-            //TODO: other construction
+            _connectors = new SqlConnector[MaxPoolSize];
+
+            // We enforce Max Pool Size, so no need to to create a bounded channel (which is less efficient)
+            // On the consuming side, we have the multiplexing write loop but also non-multiplexing Rents
+            // On the producing side, we have connections being released back into the pool (both multiplexing and not)
+            var idleChannel = Channel.CreateUnbounded<SqlConnector?>();
+            _idleConnectorReader = idleChannel.Reader;
+            _idleConnectorWriter = idleChannel.Writer;
+
+            //TODO: initiate idle lifetime and pruning fields
         }
 
         /// <inheritdoc/>
-        internal override ValueTask<SqlConnector> GetInternalConnection(SqlConnectionX owningConnection, TimeSpan timeout, bool async, CancellationToken cancellationToken)
+        internal override async ValueTask<SqlConnector> GetInternalConnection(SqlConnectionX owningConnection, TimeSpan timeout, bool async, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            CheckDisposed();
+
+            if (TryGetIdleConnector(out SqlConnector? connector))
```

Comment:
I am curious, why is TryGet a better pattern than using `SqlConnector? connector = GetIdleConnector() ` ?

Reply:
I actually do prefer that. It brings the signature in line with the other methods.